dotCMS/src/main/java/com/dotmarketing/viewtools/BrowserAPI.java

```diff
@@ -456,18 +458,25 @@ public WfData(Contentlet contentlet, List<Integer> permissions, User user, boole
 					fileAsset.getParent() != null ? fileAsset
 							.getParent() : "");
 			fileMap.put("isContentlet", false);
-			// END GRAZIANO issue-12-dnd-template
-			if (contentlet != null) {
-				fileMap.put("identifier", contentlet.getIdentifier());
-				fileMap.put("inode", contentlet.getInode());
-				fileMap.put("isLocked", contentlet.isLocked());
-				fileMap.put("isContentlet", true);
-				//Add Language Attributes to FileMap, required in Website Browser
-                Language lang = APILocator.getLanguageAPI().getLanguage(contentlet.getLanguageId());
-                fileMap.put("languageId", contentlet.getLanguageId());
-                fileMap.put("languageCode", lang.getLanguageCode());
-                fileMap.put("countryCode", lang.getCountryCode());
-			}
+            // END GRAZIANO issue-12-dnd-template
+            Language lang = null;
+            
+            if (contentlet != null) {
+                fileMap.put("identifier", contentlet.getIdentifier());
+                fileMap.put("inode", contentlet.getInode());
+                fileMap.put("isLocked", contentlet.isLocked());
+                fileMap.put("isContentlet", true);
+                lang = langAPI.getLanguage(contentlet.getLanguageId());
+                fileMap.put("languageId", lang.getId());
+            }
+            else{
+                //This is a Legacy File. Add the default Language to FileMap
+                lang = langAPI.getDefaultLanguage();
+                fileMap.put("languageId", lang.getId());
```

Comment:
fileMap.put("languageId", lang.getId()); is a common denominator, you can extract it as languageCode and countryCode.